vulcanai2/models/dnn.py

```diff
@@ -157,8 +157,10 @@ def _create_network(self, **kwargs):
         self.network = nn.Sequential(dense_layers)
 
         if self._num_classes:
-            self.network.add_module(
-                'flatten', FlattenUnit())
+            if self._get_out_dim is not None and \
```

Comment:
Why add this condition?

Reply:
it looks like `flatten` layer gets added for all use-cases. We wouldn't need flatten layer when all our `input_networks` are `DenseNet`s

Reply:
might as well check if the input network `isinstance` of a `DenseNet` no? it's what you're indirectly checking

Reply:
hmm I could add a check for if all the input_networks are `DenseNet`s but why not just check the shape of the input_tensor after merging the networks? I feel thats much simpler, no?

Reply:
1) you can't check if a function is not None i.e. `if self._get_out_dim is not None`, I think you mean to call the actual function instead of the reference to it. I'm not sure which is slower, to have a redundant flat call or check a condition. would you be able to speed test it? I think that just having the redundant flatten layer is just as fast

Reply:
>I think that just having the redundant flatten layer is just as fast

Probably, but this adds a flatten layer unnecessarily to the graph which might lead to confusions, don't you think so?

Reply:
I will take a look into speed testing

Reply:
not much of a difference in speed. The shape checking was approx faster by .05 seconds in tests, I switched to instance checking of `input_networks` as you suggested 👍